client/boot/app.js

```diff
@@ -0,0 +1,59 @@
+// Initialize localStorage polyfill before any dependencies are loaded
+require( 'lib/local-storage' )();
+if ( process.env.NODE_ENV === 'development' ) {
+	require( 'lib/wrap-es6-functions' )();
+}
+
+/**
+ * External dependencies
+ */
+import * as common from './common';
+import debugFactory from 'debug';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import createReduxStoreFromPersistedInitialState from 'state/initial-state';
+import detectHistoryNavigation from 'lib/detect-history-navigation';
+import userFactory from 'lib/user';
+
+const debug = debugFactory( 'calypso' );
+
+const withProject = () => {
+	// TODO: make project name dynamic based on config
+	const project = require( './project/wordpress-com' );
+
+	return ( funcName, ...params ) => {
+		common[ funcName ]( ...params );
+		if ( project[ funcName ] !== undefined ) {
+			project[ funcName ]( ...params );
+		}
+	};
+};
+
+const boot = currentUser => {
+	debug( "Starting Calypso. Let's do this." );
+
+	const callWithProject = withProject();
```

Comment:
@ehg @mtias this is still imperfect, but I come up with a way to call common and project specific functionalities together. My initial idea is to call project specific code as post hook. This is similar to what we do in WP core :)